Kconfig

```diff
@@ -2774,6 +2774,7 @@ source "libs/libc/Kconfig"
 source "libs/libm/Kconfig"
 source "libs/libxx/Kconfig"
 source "libs/libdsp/Kconfig"
+source "libs/libbuiltin/Kconfig"
```

Comment:
why not recompile llvm clang based on NuttX? If the system features is bound to the compiler, the compiler should be recompiled based on NuttX

Reply:
The library in compiler-rt is independent. For example, if certain functions are enabled, the functions inserted by the compiler into the program need to call the implementation in this library. In the LLVM-embedded-toolchain-for-Arm project, only libclang_rt.builtins_xxx.a is supported. If other functions are needed, such as we want to enable clang's gcov, the profile in compiler-rt needs to provide function implementations, but the arm llvm project does not support the rt.profile library, so we need to pull the library into nuttx and compile it like libcxx, similar to the patch https://github.com/apache/nuttx/pull/14327. It is not necessary to compile the compiler on nuttx.

Reply:
It would be better to make NuttX to an official target in llvm, so that llvm clang will be bound to the operating system. I am worried that the current implement will have runtime ABI functions compatibility issues on the llvm toolchain.